File: _archive/root-scripts/check_diet.js

const { createClient } = require('@supabase/supabase-js');
const dotenv = require('dotenv');
const path = require('path');

// Load .env from root folder
dotenv.config({ path: path.join(__dirname, '.env') });

const url = 'https://sdgrkwbofvxloglbumzy.supabase.co';
const key = process.env.SUPABASE_ANON_KEY;

const supabase = createClient(url, key);

async function checkDiet() {
    // Search by full_name instead
    const { data: profiles, error: pError } = await supabase
        .from('profiles')
        .select('id, full_name')
        .ilike('full_name', '%Chaya%');

    if (pError || !profiles || profiles.length === 0) {
        console.error('Profile Chaya not found:', pError || 'No results');
        return;
    }

    const traineeId = profiles[0].id;
    const fullName = profiles[0].full_name;
    console.log(`\n========================================`);
    console.log(`DIET CHECK FOR: ${fullName} (${traineeId})`);
    console.log(`========================================`);

    // 2. Get Group
    const { data: gm } = await supabase
        .from('group_members')
        .select('group_id')
        .eq('trainee_id', traineeId)
        .maybeSingle();

    const groupId = gm?.group_id;
    console.log('Group ID:', groupId || 'NOT IN ANY GROUP');

    const days = ['monday', 'tuesday', 'wednesday', 'thursday', 'friday', 'saturday', 'sunday'];
    const today = new Date().toLocaleDateString("en-US", { weekday: "long" }).toLowerCase();

    for (const day of days) {
        const isToday = (day === today);
        console.log(`\n--- ${day.toUpperCase()} ${isToday ? '[TODAY]' : ''} ---`);

        // 3. Check Custom Diet
        const { data: custom } = await supabase
            .from('custom_trainee_diet')
            .select('diet_id')
            .eq('trainee_id', traineeId)
            .eq('day_name', day)
            .maybeSingle();

        if (custom?.diet_id) {
            console.log(`  âœ… Custom Diet ID: ${custom.diet_id}`);
        } else {
            console.log('  â Œ No Custom Diet');
        }

        if (groupId) {
            const { data: groupWeekly } = await supabase
                .from('group_weekly_diet')
                .select('diet_id')
                .eq('group_id', groupId)
                .eq('day_name', day)
                .maybeSingle();

            if (groupWeekly?.diet_id) {
                console.log(`  âœ… Group Diet ID: ${groupWeekly.diet_id}`);
            } else {
                console.log('  â Œ No Group Diet (inherited)');
            }
        }
    }
}

checkDiet().catch(err => console.error(err));

File: _archive/backend-scripts/check_diet.py

import os
import uuid
from supabase import create_client, Client
from dotenv import load_dotenv

load_dotenv()

url = "https://sdgrkwbofvxloglbumzy.supabase.co"
key = os.getenv("SUPABASE_SERVICE_ROLE_KEY")
supabase: Client = create_client(url, key)

def check_diet():
    # Find trainee by email or name
    email = "chaya22791@gmail.com"
    profiles = supabase.table("profiles").select("id, full_name").eq("email", email).execute()
    
    if not profiles.data:
        print("Profile not found for email:", email)
        return

    trainee_id = profiles.data[0]['id']
    full_name = profiles.data[0]['full_name']
    print(f"Checking diet for {full_name} ({trainee_id})")

    # Get Group ID
    gm = supabase.table("group_members").select("group_id").eq("trainee_id", trainee_id).maybe_single().execute()
    group_id = gm.data['group_id'] if gm.data else None
    print(f"Group ID: {group_id}")

    days = ["monday", "tuesday", "wednesday", "thursday", "friday", "saturday", "sunday"]
    
    for day in days:
        print(f"\n--- Day: {day} ---")
        
        # Custom Diet
        custom = supabase.table("custom_trainee_diet").select("diet_id").eq("trainee_id", trainee_id).eq("day_name", day).maybe_single().execute()
        if custom.data:
            print(f"Custom Diet ID: {custom.data['diet_id']}")
        else:
            print("No Custom Diet")

        # Group Diet
        if group_id:
            group_weekly = supabase.table("group_weekly_diet").select("diet_id").eq("group_id", group_id).eq("day_name", day).maybe_single().execute()
            if group_weekly.data:
                print(f"Group Diet ID: {group_weekly.data['diet_id']}")
            else:
                print("No Group Diet")

if __name__ == "__main__":
    check_diet()
